staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -40,6 +50,40 @@ func handleError(w http.ResponseWriter, r *http.Request, err error) {
 	glog.Errorf(err.Error())
 }
 
+// requestWatermark is used to trak maximal usage of inflight requests.
+type requestWatermark struct {
+	sync.Mutex
```

Comment:
I think we have a convention to not inheirt directly from sync.Mutex. So let's change to:
lock sync.Mutex

Reply:
Hmm... I thought that at some point we had one that said that we should do it for simple things like this.

Done.

Reply:
Hmm, maybe I'm not aware of something..
But in any case, I think what you have now is definitely good too.